src/__tests__/keyboard/index.ts

```diff
@@ -0,0 +1,137 @@
+import userEvent from '../../index'
+import {addListeners, setup} from '../helpers/utils'
+
+it('type without focus', () => {
+  const {element} = setup('<input/>')
+  const {getEventSnapshot} = addListeners(document.body)
+
+  userEvent.keyboard('foo')
+
+  expect(element).toHaveValue('')
+  expect(getEventSnapshot()).toMatchInlineSnapshot(`
+    Events fired on: body
+
+    body - keydown: f (102)
+    body - keypress: f (102)
+    body - keyup: f (102)
+    body - keydown: o (111)
+    body - keypress: o (111)
+    body - keyup: o (111)
+    body - keydown: o (111)
+    body - keypress: o (111)
+    body - keyup: o (111)
+  `)
+})
+
+it('type with focus', () => {
+  const {element} = setup('<input/>')
+  const {getEventSnapshot} = addListeners(document.body)
```

Comment:
Curious why this isn't:

```suggestion
  const {element, getEventSnapshot} = setup('<input/>')
```

Reply:
Because `keyboard` does not move the focus.
If the focus is not on the `input`, there are no events on the input.
The eventSnapshot on `document.body` tells the full story for these tests and provides a meaningful error message both for wrong events and wrong event targets.

Reply:
Ah, yeah, that makes sense 👍